packages/wonder-stuff-sentry/src/kind-sentry-error.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+import {KindError, Errors} from "@khanacademy/wonder-stuff-core";
+import type {Metadata} from "@khanacademy/wonder-stuff-core";
+import {sentryDataReducer} from "./sentry-data-reducer.js";
+import type {SentryData} from "./types.js";
+
+const EmptySentryData: $ReadOnly<SentryData> = Object.freeze({
+    tags: {},
+    contexts: {},
+    fingerprint: [],
+});
+
+/**
+ * Options for constructing a `KindError`.
+ */
+type Options = {|
+    /**
+     * An error responsible for the error being created.
+     *
+     * @type {?Error}
+     */
+    cause?: ?Error,
+
+    /**
+     * Data to be attached to the error that would be reported to Sentry.
+     *
+     * @type {?$ReadOnly<$Partial<SentryData>>}
+     */
+    sentryData?: ?$ReadOnly<$Partial<SentryData>>,
```

Comment:
Using the new Flow `$Partial` type :smile:

Reply:
This allows callers to only specify a portion of the expected sentry info, even though we always assert a complete set (even if they are empty objects) elsewhere. I did this because I feel that it makes code easier to work with if we don't need null checks first.